### PR TITLE
Removed KUI usage in maps_legacy plugin

### LINKED ISSUE #3829: Remove KUI usage in `maps_legacy`
The `maps_legacy` plugin uses a single `.kui*` namespaced class. See https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards+kui+NOT+trackUI+path%3A%2F%5Esrc%5C%2Fplugins%5C%2Fmaps_legacy%2F&type=code

## To resolve

Investigate and inspect the applied CSS of that class and the rendered component. If necessary:

1. Document CSS rules provided by these classes
2. Replace these components with OUI components

Note that `maps_legacy` is a plugin dependency of the `tile_map` and `vis_type_vega` plugins, so this change should be tested in both of those cases.

### PR DESCRIPTION
**Pull request by miamia1999**
**Description**
KUI usage in maps_legacy plugin is removed
tile_map and vis_type_vega plugins were checked for functionality after changes were made in maps_legacy plugin

checking tile_map plugin for functionality: 

playground_coordinate_maps

local: 
local_coordinate_maps 

checking vis_type_vega plugin for functionality: 
playground_vega

local: 
local_vega

checking maps_legacy plugin for functionality after changes:
playground_region_maps

local:
local_region_maps

**Issues Resolved**
closes #3829 
In a team with @curq 

### Check List

- [ ] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/maps_legacy/public/map/opensearch_dashboards_map.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ import { i18n } from '@osd/i18n';
 import { ORIGIN } from '../common/constants/origin';
 import { getToasts } from '../opensearch_dashboards_services';
 import { L } from '../leaflet';
+import './_legend.scss';
```
Comment (BSFishy) [inline src/plugins/maps_legacy/public/map/opensearch_dashboards_map.js:44]: Shouldn't this already be imported by `src/plugins/maps_legacy/public/map/index.scss`?

Comment (miamia1999) [reply src/plugins/maps_legacy/public/map/opensearch_dashboards_map.js]: Thank you, @BSFishy for your review. I have removed the redundant import in the latest commit

**Review by AMoo-Miki — APPROVED** [review]
I wish we could have used OUI here, perhaps with the `vector` icon but this a good first step.

**Comment by ananzh** [comment]
@miamia1999 I think all the changes are good. I could approve it after you update the CHANGELOG file. Thank you for the contribution and detail explains.

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,7 +39,6 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 - [Doc] Add COMMUNICATIONS.md with info about Slack, forum, office hours ([#3837](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3837))
 - [Saved Object Service] Adds design doc for new Saved Object Service Interface for Custom Repository [#3954](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3954)
```
Comment (joshuarrrr) [inline CHANGELOG.md:41]:
```suggestion
- [Saved Object Service] Adds design doc for new Saved Object Service Interface for Custom Repository [#3954](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3954)
- Add plugin development section in DEVELOPER_GUIDE.md ([#778](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3989))
```

File: src/plugins/maps_legacy/public/map/_legend.scss [hunk 3]
```diff
@@ -33,3 +33,12 @@
   left: $euiSizeXXL;
   white-space: nowrap;
 }
+
+.mapButton {
+  display: inline-block;
+  font: normal normal normal 14px/1 FontAwesome, sans-serif;
```
Comment (joshuarrrr) [inline src/plugins/maps_legacy/public/map/_legend.scss:39]: @AMoo-Miki - for a follow-up PR, is there a way we can reference a particular OUI icon directly via SASS/CSS for a case like this?

**Review by joshuarrrr — COMMENTED** [review]
Just some unintentional changelog entries from the conflict resolution. Otherwise looks good.

**Review by joshuarrrr — DISMISSED** [review]
I resolved the remaining changelog issues.

File: src/plugins/maps_legacy/public/map/_legend.scss (hunk 3 above)
Comment (AMoo-Miki) [reply src/plugins/maps_legacy/public/map/_legend.scss:39]: This piece of code is not written the React way else they could have just used `OuiIcon`. There is no direct access to the imagery.

Comment (joshuarrrr) [reply src/plugins/maps_legacy/public/map/_legend.scss:39]: Yeah, I figured. I'm not sure we'd even want to expose direct access to the assets as part of OUI.

**Review by joshuarrrr — DISMISSED** [review]

**Review by ananzh — DISMISSED** [review]

**Review by BSFishy — APPROVED** [review]

**Comment by joshuarrrr** [comment]
All checks pass.

**Review by joshuarrrr — APPROVED** [review]

**Review by BSFishy — APPROVED** [review]

**Pull request merged**